plugin/vm_processor.go

```diff
@@ -0,0 +1,128 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/resources"
+	"golang.org/x/sync/errgroup"
+)
+
+// vmJob represents a single VM processing job
+type vmJob struct {
+	resource      resources.GenericResourceExpanded
+	resourceGroup string
+	name          string
+}
+
+func (p *AzurePlugin) processStandardVMs(ctx context.Context, resourceInfos []resources.GenericResourceExpanded,
+	clients *azureClients, vmToNetworkMap map[string]networkInfo) error {
+
+	// Create error group for worker management
+	g, ctx := errgroup.WithContext(ctx)
+
+	// Create job channel
+	jobs := make(chan vmJob)
+
+	// Start workers
+	workerCount := constDefaultWorkers
+	if len(resourceInfos) < workerCount {
+		workerCount = len(resourceInfos)
+	}
```

Comment:
Did you find this was beneficial in your testing? I think the Go runtime should be handle hundreds of thousand of goroutines at once (though each one has a memory cost). I might suggest setting constDefaultWorkers to 1000 or something and it should still be very cheap.

Reply:
I see that we're using this same constant in the other parallel jobs - if you really want to limit parallelism (which, again, I'm not sure we really need to), you might want to pass around a simple channel based semaphore (something like this: https://jbrandhorst.com/post/go-semaphore/)

Reply:
Honestly, didn't attempt this with just a default worker value It just seemed common sense to limit workers to only the number I actually needed if it was less than the default worker pool size. However, if they are as cheap as you say there's not going to be any problem with setting a sensible common value in leaving it at that.